scalafix-reflect/src/main/scala/scalafix/internal/reflect/RuleInstrumentation.scala

```diff
@@ -15,26 +15,26 @@ object RuleInstrumentation {
       def unapply(templ: Template): Boolean = templ match {
 
         // v0
-        case Template(_, init"Rewrite" :: _, _, _) => true
-        case Template(_, init"Rule($_)" :: _, _, _) => true
-        case Template(_, init"SemanticRewrite($_)" :: _, _, _) => true
-        case Template(_, init"SemanticRule($_, $_)" :: _, _, _) => true
+        case Template(_, Init(Type.Name("Rewrite"), _, _) :: _, _, _) => true
+        case Template(_, Init(Type.Name("Rule"), _, List(List(_))) :: _, _, _) => true
+        case Template(_, Init(Type.Name("SemanticRewrite"), _, List(List(_))) :: _, _, _) => true
+        case Template(_, Init(Type.Name("SemanticRule"), _, List(List(_, _))) :: _, _, _) => true
```

Comment:
I run it in a worksheet and got:
```

import scala.meta._
init"Rewrite"
// res0: Init = Init(tpe = Type.Name(value = "Rewrite"), name = _, argss = List())
init"Rule(a)"
// res1: Init = Init(tpe = Type.Name(value = "Rule"), name = _, argss = List(List(Term.Name(value = "a"))))
init"SemanticRewrite(a)"
// res2: Init = Init(tpe = Type.Name(value = "SemanticRewrite"), name = _, argss = List(List(Term.Name(value = "a"))))
init"SemanticRule(a, b)"
// res3: Init = Init(
//   tpe = Type.Name(value = "SemanticRule"),
//   name = _,
//   argss = List(List(Term.Name(value = "a"), Term.Name(value = "b")))
// )
```

So I would say that we should tweak it a little bit:
```suggestion
        case Template(_, Init(Type.Name("Rewrite"), _, Nil) :: _, _, _) => true
        case Template(_, Init(Type.Name("Rule"), _, List(List(_))) :: _, _, _) => true
        case Template(_, Init(Type.Name("SemanticRewrite"), _, List(List(_))) :: _, _, _) => true
        case Template(_, Init(Type.Name("SemanticRule"), _, List(List(_, _))) :: _, _, _) => true
```